0.4.0 release notes (#1147)
weaveworks/eksctl: docs/release_notes/0.4.0.md

# Release 0.4.0

## Features

> Includes all features of the previous version, plus:

Reviewer: How about we just rename the heading to "New Features"? This line is quite redundant.

Author: It is from a suggestion from a user in a different PR earlier this week.

> Includes all features of the  previous version, plus:

Reviewer 2: ```suggestion
Includes all features of the previous version, plus:
```

> - New experimental feature `generate profile`

Reviewer 2: We should link to the PR:
```suggestion
- New experimental feature `generate profile` (#1132)
```

> - New experimental feature `generate profile` (#1132)

Reviewer: ```suggestion
- New experimental command `generate profile` (#1132)
```
Also, do we have a link to docs?

Author: Not until we release.

> - Added installation of Helm/Tiller as part of the `install flux` operation

Reviewer 2: ```suggestion
- Added installation of Helm/Tiller as part of the `install flux` operation (#1136)
```

> - Added installation of Helm/Tiller as part of the `install flux` operation (#1136)

Reviewer: Let's use full command names, it's not clear otherwise:
```suggestion
- Added installation of Helm/Tiller as part of the `eksctl install flux` command (#1136)
```

## Bug Fixes and code improvements
- perform defaulting and validation in all commands (#1124)
- improve nodegroup validation to detect duplicate names (#1116)
- tidy up command usage formatting (#1112)
- various internal improvements (#1113, #1117, #1118, #1119)

> - remove the ability of skipping tasks (#1101)

Reviewer: I'd just add `#1101` on the line above, it's a tiny tiny thing.

- added tests for experimental features

> - refactored `install flux` experimental feature for testability

Reviewer: Also, not important to mention, the pervious item already cover this.

> - added common AWS CLI options to `install flux`

Reviewer: This is also unimportant, but if you insist on adding it do mention the PR and please don't use "AWS CLI" (which was used in PR tittle, but it's actually misleading, as "AWS CLI" stands for [something else](https://aws.amazon.com/cli/).

```suggestion
- add common AWS flags to `eksctl install flux` (#1082) 
```

Author: This comment should be on commit descriptions too, then. They mention AWS CLI.

> - cmd description spacing fixed

Reviewer: Please definitely remote this, it's just meaningless to anyone who reads this.

Author: Remove what? the 4 lines?

> - use the same kubeconfig.yaml as Amazon Linux (#994)

Reviewer: ```suggestion
- add support for kubelet config files for Ubuntu nodes (#994)
```

> ## Documentation

Reviewer: This should have links.
